components/ActionSheet/src/MDCActionSheetController.h

```diff
@@ -0,0 +1,182 @@
+ with actionSheetControllerWithTitle:message:
+ */
+- (nonnull instancetype)initWithNibName:(NSString *)nibNameOrNil
+                                 bundle:(NSBundle *)nibBundleOrNil NS_UNAVAILABLE;
+
+/**
+ Action sheet controllers must be created with actionSheetControllerwithTitle:
+   or with actionSheetControllerWithTitle:message:
+ */
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder NS_UNAVAILABLE;
+
+/**
+ Adds an action to the action sheet.
+
+ Actions are the possible reactions of the user to the presented alert. Actions are added as a
+ list item in the list of the action sheet.
+ Action buttons will be laid out from top to bottom depending on the order they
+ were added.
+
+ @param action Will be added to the end of MDCActionSheetController.actions.
+ */
+- (void)addAction:(nonnull MDCActionSheetAction *)action;
+
+/**
+ The actions that the user can take in response to the action sheet.
+
+ The order of the actions in the array matches the order in which they were added to the action sheet.
+ */
+@property (nonatomic, nonnull, readonly) NSArray<MDCActionSheetAction *> *actions;
+
+/**
+ The title of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *title;
+
+/**
+ The message of the action sheet controller.
+
+ If this is updated after presentation the view will be updated to match the new value.
+ */
+@property (nonatomic, nullable, copy) NSString *message;
+
+/*
+ Indicates whether the button should automatically update its font when the device’s
+ UIContentSizeCategory is changed.
+
+ This property is modeled after the adjustsFontForContentSizeCategory property in the
+ UIConnectSizeCategoryAdjusting protocol added by Apple in iOS 10.0.
+
+ If set to YES, this button will base its text font on MDCFontTextStyleButton.
+
+ Defaults value is NO.
+ */
+@property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
+    BOOL mdc_adjustsFontForContentSizeCategory;
+
+@end
+
+/**
+ MDCActionSheetActionHandler is a block that will be invoked when the action is selected.
+ */
+typedef void (^MDCActionSheetHandler)(MDCActionSheetAction *_Nonnull action);
+
+/**
+ An instance of MDCActionSheetAction is passed to MDCActionSheetController to add an
+ action to the action sheet.
+ */
+@interface MDCActionSheetAction : NSObject <NSCopying>
+
+
+/**
+ Returns an action sheet action with the populated given values.
+
+ @param title The title of the list item shown in the list
+ @param image The icon of the list item shown in the list
+ @param handler A block to execute when the user selects the action.
+ @return An initialized MDCActionSheetAction object.
+ */
++ (nonnull instancetype)actionWithTitle:(nonnull NSString *)title
+                                  image:(nonnull UIImage *)image
+                                handler:(__nullable MDCActionSheetHandler)handler;
```

Comment:
What's the idea for a nullable block here? Do we have an expected use case for an action sheet with buttons that do nothing?

Reply:
I don't know why a client would have a button that does nothing but I think it is a possibility. I don't see a benefit in marking it as nonnull.

Reply:
I think the benefit is to help prevent programmer errors. Unless we have a use case, I would think attaching an action that does nothing to an action sheet should be considered incorrect.

Reply:
What about an action that "dismisses" the sheet?

Reply:
That should be provided in the sheet's app bar (guidelines excerpt below).  Even if there were an explicit Dismiss/Cancel option, I would expect that is has an action to dismiss the action sheet.  Are you thinking that the ActionSheet would dismiss itself if any list item without an attached handler were activated?

> Full-screen modal bottom sheets contain a dismiss icon that dismisses the sheet entirely from view.
> &mdash; https://material.io/design/components/sheets-bottom.html#modal-bottom-sheet